src/main/webapp/app/entities/lecture/lecture-detail.component.ts

```diff
@@ -1,23 +1,60 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { SafeHtml } from '@angular/platform-browser';
 import { ActivatedRoute } from '@angular/router';
-import { Lecture } from 'app/entities/lecture';
+import { HttpResponse } from '@angular/common/http';
+import { Subscription } from 'rxjs/Subscription';
+import { JhiEventManager } from 'ng-jhipster';
+
+import { Lecture, LectureService } from 'app/entities/lecture';
+import { ArtemisMarkdown } from 'app/components/util/markdown.service';
 
 @Component({
     selector: 'jhi-lecture-detail',
     templateUrl: './lecture-detail.component.html',
 })
-export class LectureDetailComponent implements OnInit {
+export class LectureDetailComponent implements OnInit, OnDestroy {
     lecture: Lecture;
 
-    constructor(protected activatedRoute: ActivatedRoute) {}
+    formattedDescription: SafeHtml | null;
+
+    private subscription: Subscription;
+    private eventSubscriber: Subscription;
+
+    constructor(
+        protected activatedRoute: ActivatedRoute,
+        private eventManager: JhiEventManager,
+        private lectureService: LectureService,
+        private artemisMarkdown: ArtemisMarkdown,
+    ) {}
 
     ngOnInit() {
         this.activatedRoute.data.subscribe(({ lecture }) => {
             this.lecture = lecture;
         });
+        this.subscription = this.activatedRoute.params.subscribe(params => {
+            this.load(params['id']);
+        });
+        this.registerChangeInLecture();
+    }
+
+    load(id: number) {
+        this.lectureService.find(id).subscribe((lectureResponse: HttpResponse<Lecture>) => {
+            this.lecture = lectureResponse.body!;
```

Comment:
I added a lot of changes here which I thought were necessary because of the formattedDescription variable. Now that I am using the htmlForMarkdown pipe in the lecture-detail.component.html file, I would reverse all the changes I made in this one (except chnging the scope of the activatedRoute to private).  Or do you believe Ithey are necessary?

Reply:
Sorry, I didn't understand your approach at first. Now I checked that you actually don't need to create the subscriptions at all. If for some reason you don't receive lecture from the route data, you need to check the resolve method in `lecture.route`. 
I also can't reproduce the bug you mentioned on the test server. Can you share additional details needed to reproduce it?

Reply:
According to the bug: when I reverse the change I made on row 33 in the lecture-detail.component.html, locally I get an error in the browser console when I click on  Course Administration -> Lectures (on an existing course) -> View (the view button does not have a function).